### ISSUE TITLE
Academic Units functionality review

### ISSUE THREAD

**Issue by boonebgorges** [cuny-academic-commons/openlab-theme#208]
From @charliemedwards 

![screenshot_2018-05-21_15-20-46](https://user-images.githubusercontent.com/246627/40315333-96aa1d28-5d0a-11e8-8963-6d6c92c077b1.png)

@bree-z This was previously signed-off on. Could you have another look to see if you can confirm?

**Comment by bree-z** [comment]
I'm able to add a second top-level unit with children (so we have Schools > Departments, and Units > Offices) but I did encounter one strange thing.  

- I created a new top-level unit called Units.
- I added a new Unit within this called "Unit 1"
- I created a child of Units called Offices.  
- I created an Office called "Office of Flotsum" and in the Parent box, I chose "Unit 1", but after choosing the unit, the dropdown appeared blank.  

<img width="1085" alt="screen shot 2018-06-06 at 1 07 01 pm" src="https://user-images.githubusercontent.com/7189824/41032509-44d9b3b2-698c-11e8-9a1b-44c6394a650f.png">

- I clicked "Add New" 
- "Office of Flotsum" appeared in the listing of offices, but without a parent.  
- On the front end, if I go to create a project, "office of Flotsum" appears as a choice after I choose Unit 1 as the parent unit.  
- On the back end, if I edit "Office of Flotsum", I can choose "Unit 1" as the parent and save it. It saves correctly and "Unit 1" appears as the parent. 
- I created one more office called "Office of Jetsum" and experience the same thing where I couldn't  add Unit 1 as the parent.  When I edit it, this is what I see, which appears to be related to #126 

<img width="1045" alt="screen shot 2018-06-06 at 1 18 48 pm" src="https://user-images.githubusercontent.com/7189824/41032496-3ba76dca-698c-11e8-898d-856263029bf4.png">

And here's what I see on the front end, which is all schools, departments, units, and offices, as intended:

<img width="703" alt="screen shot 2018-06-06 at 1 22 34 pm" src="https://user-images.githubusercontent.com/7189824/41032656-b969206e-698c-11e8-8935-714db4afecfb.png">

**Comment by boonebgorges** [comment]
@bree-z To clarify, all of the steps you bullet-pointed above were taken without any page refreshes, correct? The fact that it's working as expected on the front end suggests that this is just a problem with the back-end app, not the user-facing functionality.

**Comment by bree-z** [comment]
Yes, I believe so.  

I just logged back in to test one more time, and was able to add a new Office ("More Admins"), and this time I could choose "Unit 1" in the Parent dropdown, and have "stick" so that "More Admins" was added to the list of offices, with the parent of Unit 1, without having to edit it to add the parent.  

I opened up My Projects in a new tab, created a new project, and saw it listed there as a child of Unit 1, together with the other Offices I'd previously created.

**Comment by boonebgorges** [comment]
Thanks, Bree. I spent some time trying to figure this out, but can't see how to fix it without rebuilding the entire application. As a workaround, I'm now forcing a page refresh when you modify an Academic Unit Type, which seems to clear up most of these problems. Could you please have a look?

**Comment by bree-z** [comment]
Thanks, Boone!  This now looks good to me in CBOX-OL Dashboard > Academic Units.  

However (sorry!), I discovered one more issue:

- I added new parent and children units (e.g. Programs > Certifications) 
- I added them to an existing group
- They appeared as choices in the group settings, but after I applied them, the labels didn't appear on the group profile.  For example, in this group: http://openlabdev.commonsinabox.org/groups/back-from-vacation-blues/ 

![missingunitlabels](https://user-images.githubusercontent.com/7189824/41725588-d7d4feae-7578-11e8-9fa1-72c77c3ecb75.png)


- I went back to the Dashboard > Academic Units settings page and changed the plural labels (which were singular by default, e.g. Plural of Program was "Program")
- I created a new group (http://openlabdev.commonsinabox.org/groups/adding-academic-units/) and all the labels appeared on the group profile. 
- When I checked the other group linked above, the labels appeared there too.  
- Then, I deleted the Program/Certificate units and created new ones, "Parent Unit" and "Child Unit"
- I created a new group: http://openlabdev.commonsinabox.org/groups/more-academic-units/
- This time, just the label for the Child Unit was missing.  

<img width="1218" alt="missingunitlabel" src="https://user-images.githubusercontent.com/7189824/41725592-dba47bcc-7578-11e8-9c21-c552173179b2.png">

- As above, I went back to the Dashboard > Academic Units settings page and changed the plural label for child unit (this was also singular by default, e.g. Plural of Child Unit was "Child Unit"), and all labels appeared on the group profile.  

Thanks!

**Comment by boonebgorges** [comment]
Thanks, Bree. I've pushed up what I think is a fix.

**Comment by bree-z** [comment]
Thanks, Boone.  I think it's working.  I deleted the previous units I'd added and created new ones.  I was able to add them to a new project, an existing course, and an existing project, and they appeared on the profile, as expected.  

The only project I couldn't add them to was this one: http://openlabdev.commonsinabox.org/groups/learning-to-read/ . But, this is a really old group, from initial testing, perhaps created before you even worked on academic units development?  And, I couldn't seem to add any academic units to this one.  I'd add them, save, and they wouldn't show up.  Then I'd go back to settings and they'd all be unchecked.  

If this sounds OK, I think we can close this ticket, thanks!
